xray/aws_test.go

```diff
@@ -178,39 +203,34 @@ func testClientFailedConnection(t *testing.T, svc *lambda.Lambda) {
 	assert.NotEmpty(t, connectSubseg.Subsegments)
 }
 
-func testClientWithoutSegment(t *testing.T, svc *lambda.Lambda) {
-	Configure(Config{ContextMissingStrategy: &TestContextMissingStrategy{}})
-	defer ResetConfig()
-
-	ctx := context.Background()
-	_, err := svc.ListFunctionsWithContext(ctx, &lambda.ListFunctionsInput{})
-	assert.NoError(t, err)
+func testClientWithoutSegment(ctx context.Context, td *TestDaemon, t *testing.T, svc *lambda.Lambda) {
```

Comment:
Any particular issue with the function ? How can we fix this ?

Reply:
Configure and ResetConfig modify the environment values,
it effects the other tests which get the configure from the environment values.

see https://github.com/aws/aws-xray-sdk-go/pull/177#issuecomment-575416696

Reply:
oh, I have a misunderstanding.
Configure and ResetConfig don't modify the environment values.

But they still modify global configure. it may cause race condition.
I think we must not use them in tests.